File: Data/Data.Models/Entities/TripFilter.cs
using System;
using System.Text.Json.Serialization;

namespace Data.Data
{
    public class TripFilter
    {

        [JsonPropertyName("driverEmail")]
        public string DriverEmail { get; set; }

        [JsonPropertyName("passengerEmail")]
        public string PassengerEmail { get; set; }

        [JsonPropertyName("minimumArrivalDate")]
        public string MinimumArrivalDate { get; set; }
        
        [JsonPropertyName("maximumArrivalDate")]
        public string MaximumArrivalDate { get; set; }

        [JsonPropertyName("pickupAddress")]
        public string PickupAddress { get; set; }

        [JsonPropertyName("dropoffAddress")]
        public string DropoffAddress { get; set; }

        [JsonPropertyName("arrivalDate")]
        public string ArrivalDate { get; set; }

    }
}

File: Logic/src/models/TripFilter.java
package models;

public class TripFilter {

    private String driverEmail;
    private String passengerEmail;
    private String minimumArrivalDate;
    private String maximumArrivalDate;
    private String pickupAddress;
    private String dropoffAddress;
    private String arrivalDate;

    public TripFilter(String driverEmail, String passengerEmail, String minimumArrivalDate, String maximumArrivalDate, String pickupAddress, String dropoffAddress, String arrivalDate) {
        this.driverEmail = driverEmail;
        this.passengerEmail = passengerEmail;
        this.minimumArrivalDate = minimumArrivalDate;
        this.maximumArrivalDate = maximumArrivalDate;
        this.pickupAddress = pickupAddress;
        this.dropoffAddress = dropoffAddress;
        this.arrivalDate = arrivalDate;
    }

    public String getArrivalDate() {
        return arrivalDate;
    }

    public String getPickupAddress() {
        return pickupAddress;
    }

    public String getDropoffAddress() {
        return dropoffAddress;
    }

    public String getDriverEmail() {
        return driverEmail;
    }

    public String getPassengerEmail() {
        return passengerEmail;
    }

    public String getMinimumArrivalDate() {
        return minimumArrivalDate;
    }

    public String getMaximumArrivalDate() {
        return maximumArrivalDate;
    }
}
